client/state/preferences/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+
+export const USER_SETTING_KEY = 'calypso_preferences';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get().then( ( data ) => {
+			dispatch( {
+				type: PREFERENCES_RECEIVE,
+				data
+			} );
+		} );
+	};
+}
+
+const save = () => ( dispatch, getState ) => {
```

Comment:
Between this and #202, I'm wondering if we should explore creating a separate REST API endpoint specifically for managing Calypso preferences, or at least update the current endpoint to enable more granular updates (e.g. accept an object of key values). As you've discovered, right now it's a bit of a pain to update a single preference.